src/common/transformations/src/transformations/common_optimizations/convert_u4_weights_zero_point_to_scalar.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/convert_u4_weights_zero_point_to_scalar.hpp"
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/reference/autobroadcast_binop.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace {
+bool convert_zero_point_to_scalar(const std::shared_ptr<ov::op::v0::Constant>& weights,
+                                  const std::shared_ptr<ov::op::v0::Constant>& zero_point) {
+    auto zero_point_shape = zero_point->get_shape();
+    if (ov::shape_size(zero_point_shape) == 1)
+        return false;
+
+    const auto& weights_shape = weights->get_shape();
+    const size_t weights_rank = weights_shape.size();
+    const size_t zero_point_rank = zero_point_shape.size();
+    // Zero point constant can be converted into scalar only if this does not affect Subtract output shape
+    if (weights_rank < zero_point_rank || ov::shape_size(weights_shape) < ov::shape_size(zero_point_shape))
+        return false;
+
+    zero_point_shape.insert(zero_point_shape.begin(), weights_rank - zero_point_rank, 1);
```

Comment:
It seems "zero_point_shape.insert" is not necessary, since shape comparison can be done without insertion and zero_point_shape won't be needed after that.

Reply:
Thanks for the comment.
Please note: `ConvertU4WeightsZeroPointToScalar` transformation code (and the corresponding test) is from the PR #20503 on which this PR depends. I will transfer your comment to #20503